atecc608a_se.c

```diff
@@ -208,24 +247,68 @@ static psa_status_t atecc608a_asymmetric_sign(psa_key_slot_number_t key_slot,
     return status;
 }
 
+psa_status_t atecc608a_asymmetric_verify(psa_key_slot_number_t key_slot,
+                                         psa_algorithm_t alg,
+                                         const uint8_t *p_hash,
+                                         size_t hash_length,
+                                         const uint8_t *p_signature,
+                                         size_t signature_length)
+{
+    const uint16_t key_id = key_slot;
+    psa_status_t status = PSA_ERROR_GENERIC_ERROR;
+    bool is_verified = false;
+
+    /* Keys 8 to 15 can store public keys. */
+    if (key_id < 7 || key_id > 15)
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    /* We can only do ECDSA on SHA-256 */
+    if (alg != PSA_ALG_ECDSA(PSA_ALG_SHA_256) && alg != PSA_ALG_ECDSA_ANY)
```

Comment:
This check is redundant with the similar check at import time: the core is guaranteed to pass an algorithm that complies with the policy. You can leave it in for robustness, or omit it to save a couple of bytes of code size.

Reply:
How will the core know what policy the driver/hardware is physically capable of implementing? Shouldn't the driver be the source of this information and be able to reject it (either at policy setting time [for which we'd need new driver function pointers] or at use time [here])?

Reply:
The driver interface needs to change to allow the driver code to pick a slot number when creating a key. We have some architecture discussions on the topic but no concrete header files yet.